private/pkg/protoencoding/reparse_unrecognized.go

```diff
@@ -22,6 +22,8 @@ import (
 // ReparseUnrecognized uses the given resolver to parse any unrecognized fields in the
 // given reflectMessage. It does so recursively, resolving any unrecognized fields in
 // nested messages.
+//
+// Deprecated: Use ReparseExtensions instead.
```

Comment:
Is there a specific reason we need to keep this around? If it's still used in core, refactor core to not use it

Reply:
I think description answers this - yes, force other repos to refactor

Reply:
Done